pkg/sql/run_control_test.go

```diff
@@ -848,6 +848,54 @@ func TestIdleInTransactionSessionTimeoutCommitWaitState(t *testing.T) {
 	}
 }
 
+func TestStatementTimeoutRetryableErrors(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	ctx := context.Background()
+
+	numNodes := 1
+	tc := serverutils.StartTestCluster(t, numNodes,
+		base.TestClusterArgs{
+			ReplicationMode: base.ReplicationManual,
+		})
+	defer tc.Stopper().Stop(ctx)
+
+	conn, err := tc.ServerConn(0).Conn(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = conn.QueryContext(ctx,
+		`SET statement_timeout = '0.1s'`)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	timer := time.AfterFunc(1*time.Second, func() {
```

Comment:
nit: could you add a comment here mentioning that this is here to verify that the timeout error is returned before the 2 second interval passed to `force_retry` elapses?